packages/libs/deer/desert/src/token.rs

```diff
@@ -3,69 +3,202 @@ use core::fmt::{Debug, Display, Formatter};
 use deer::Number;
 
 // TODO: test
-#[derive(Debug, Copy, Clone, PartialEq)]
+// TODO: this should be `Copy`, but `Number` has no &'static constructor
+#[derive(Debug, Clone, PartialEq)]
 pub enum Token {
-    /// A serialized `bool`
+    /// A serialized [`bool`]
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&true, &[Token::Bool(true)])
+    /// ```
+    Bool(bool),
+
+    /// A serialized [`Number`]
+    ///
+    /// ```
+    /// use deer::Number;
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&Number::from(1), &[Token::Number(Number::from(1))])
+    /// ```
+    Number(Number),
+
+    /// A serialized [`u128`]
+    ///
+    /// this is a separate variant, as [`Number`] is unable to fully represent 128 bit values, this
+    /// means that u128/i128 need special support.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1u128, &[Token::U128(1)])
+    /// ```
+    U128(u128),
+
+    /// A serialized [`i128`]
+    ///
+    /// this is a separate variant, as [`Number`] is unable to fully represent 128 bit values, this
+    /// means that u128/i128 need special support.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1i128, &[Token::I128(1)])
+    /// ```
+    I128(i128),
+
+    /// A serialized [`usize`]
+    ///
+    /// this is a separate variant, because there is no guarantee about the width of a usize/isize,
+    /// it depends on the size to reference a memory address on the host system. There might be
+    /// systems that use 128 bit in the future that rusts supports, which means we too need to
+    /// support those, by special casing usize.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1usize, &[Token::USize(1)])
+    /// ```
+    USize(usize),
+
+    /// A serialized [`isize`]
+    ///
+    /// this is a separate variant, because there is no guarantee about the width of a usize/isize,
+    /// it depends on the size to reference a memory address on the host system. There might be
+    /// systems that use 128 bit in the future that rusts supports, which means we too need to
+    /// support those, by special casing usize.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1isize, &[Token::ISize(1)])
+    /// ```
+    ISize(isize),
+
+    /// A serialized [`char`]
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&'a', &[Token::Char('a')])
+    /// ```
+    Char(char),
```

Comment:
How come we support char as a standalone token?

Reply:
`char` is a primitive type (and copy), and there may be some deserializers that support `char` as a separate type. The default anyway is that `visit_char` just defers to `visit_str`.

(This is also here because serde already supports this type, and I think in general it is a good idea to cover every primitive type that exists)